lib/ansible/modules/network/avi/avi_api_session.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python
+"""
+# Created on Aug 12, 2016
+#
+# @author: Gaurav Rastogi (grastogi@avinetworks.com) GitHub ID: grastogi23
+#
+# module_check: not supported
```

Comment:
It does support impotency fully. However, it does not support module_check as a dry-run because of limitation on Avi API which do not have a dry-run option. Checking will alter state on the Avi controllers.